caikit/runtime/server_base.py

```diff
@@ -39,6 +47,54 @@ def __init__(self, base_port: int, tls_config_override: Optional[aconfig.Config]
         )
         log.debug4("Full caikit config: %s", get_config())
 
+        # Configure using the log level and formatter type specified in config.
+        caikit.core.toolkit.logging.configure()
+
+        # We should always be able to stand up an inference service
+        self.inference_service: ServicePackage = (
+            ServicePackageFactory.get_service_package(
+                ServicePackageFactory.ServiceType.INFERENCE,
+            )
+        )
+
+        # But maybe not always a training service
+        try:
+            training_service: Optional[
+                ServicePackage
+            ] = ServicePackageFactory.get_service_package(
+                ServicePackageFactory.ServiceType.TRAINING,
+            )
+        except CaikitRuntimeException as e:
+            log.warning("Cannot stand up training service, disabling training: %s", e)
+            training_service = None
+
+        self.training_service = training_service
+
+    @classmethod
+    def _start_metrics_server(cls) -> None:
+        """Start a single instance of the metrics server based on configuration"""
+        if not cls._metrics_server_started and get_config().runtime.metrics.enabled:
+            log.info(
+                "Serving prometheus metrics on port %s",
+                get_config().runtime.metrics.port,
+            )
+            with alog.ContextTimer(log.info, "Booted metrics server in "):
+                start_http_server(get_config().runtime.metrics.port)
+            cls._metrics_server_started = True
+
+    def _intercept_interrupt_signal(self) -> None:
```

Comment:
Per comment in `__main__`, I think this should probably move out of this base class into the `main` functions of the respective server modules

Reply:
I have it in the base class to just avoid duplication of code, but this is called from within `main` functions of individual servers. Does that make sense code wise?